ecl/hql/hqlcache.cpp

```diff
@@ -340,6 +342,161 @@ void convertSelectsToPath(StringBuffer & filename, const char * eclPath)
+                                {
+                                    attrs.append(*(LINK(attr)));
+                                }
+                                else
+                                {
+                                    IValue * val = attr->queryValue();
+                                    if (!val)
+                                        throw MakeStringExceptionDirect(0, "Cannot simplify record");
+                                    if (val->getTypeCode()==type_int)
+                                        attrs.append(*createConstant(val->getIntValue()));
+                                    else
+                                        attrs.append(*createConstant(LINK(val)));
+                                    same2 = false;
+                                }
+                            }
+                            if (same2)
+                            {
+                                children.append(*(LINK(cur)));
+                            }
+                            else
+                            {
+                                OwnedHqlExpr newattr = createExprAttribute(cur->queryName(),attrs);
+                                children.append(*(newattr.getClear()));
+                                same = false;
+                            }
+                        }
+                    }
+                    else
+                        same = false;
+                }
+                OwnedITypeInfo newType = transformType(type);
+
+                if (type != newType)
+                    return createField(expr->queryId(), newType.getClear(), children);
+
+                if (same)
+                    return LINK(expr);
+                return expr->clone(children);
+            }
+        case no_param:
+            {
+                ITypeInfo * type = expr->queryType();
+                OwnedITypeInfo newType = transformType(type);
+                HqlExprArray children;
+                transformChildren(expr, children);      // could just unwind
+                return createParameter(expr->queryId(), UnadornedParameterIndex, newType.getClear(), children);
+            }
+        }
+
+        return QuickHqlTransformer::createTransformed(expr);
+    }
+
+protected:
+    IHqlExpression * createTransformedField(IHqlExpression * expr)
+    {
+        Owned<ITypeInfo> type = getFullyUnqualifiedType(expr->queryType());
+        OwnedITypeInfo newType = transformType(type.getClear());
+        IIdAtom * id = expr->queryId();
+        return createField(id, newType.getClear(), nullptr);
+    }
+    ITypeInfo * transformType(ITypeInfo * type)
+    {
+        switch (type->queryModifier())
+        {
+        case typemod_original:
+            {
+                return transformType(type->queryTypeBase()); //Strip all original annotations
+            }
+        case typemod_none:
+            {
+                //Ensure all records with the same format get the same original modifier
```

Comment:
Why do you want to do this?  Would it be better to not add any modifier.  This looks like it might cause problems when regenerating.

Reply:
@ghalliday Please see comment below